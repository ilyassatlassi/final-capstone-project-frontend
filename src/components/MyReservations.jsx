import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchReservations } from '../redux/slices/reservations';
import { fetchDoctors } from '../redux/slices/doctors';

const MyReservations = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchReservations());
    dispatch(fetchDoctors());
  }, [dispatch]);
  const newReservationArray = [];
  const { reservations } = useSelector((store) => store.reservations);
  const { doctors } = useSelector((store) => store.doctors);
  reservations.map((reservation) => {
    doctors.map((doctor) => {
      if (reservation.doctor_id === doctor.id) {
        const dateObject = new Date(reservation.time);
        const timeOptions = {
          hour: '2-digit',
          minute: '2-digit',
        };
        const formattedTime = dateObject.toLocaleTimeString('en-US', timeOptions);
        const newReservationObject = {
          doctor: doctor.name,
          time: formattedTime,
          date: reservation.date,
          city: reservation.city,
          doctorImage: doctor.image,
          id: reservation.id,
        };
        newReservationArray.push(newReservationObject);
      }
    });
  });

  return (
    <section className="grid lg:grid-cols-3 h-screen gap-4 w-screen justify-center p-6 bg-[#97BFOE]">
      {
      newReservationArray.map((reservation) => (
        <ul key={reservation.id} className="h-1/2 w-full">
          <li className="h-full w-full">
            <img className="h-full w-full rounded-xl" src={reservation.doctorImage} alt={reservation.doctor} />
          </li>
          <li className="flex gap-2">
            <h3 className="font-bold">
              Doctor&apos;s Name:
            </h3>
            {reservation.doctor}
          </li>
          <li className="flex gap-2">
            <h3 className="font-bold">City:</h3>
            {' '}
            {reservation.city}
          </li>
          <li className="flex gap-2">
            <h3 className="font-bold">Date of Meeting:</h3>
            {' '}
            {reservation.date}
          </li>
          <li className="flex gap-2">
            <h3 className="font-bold">Time of Meeting:</h3>
            {' '}
            {reservation.time}
          </li>
        </ul>
      ))
}
    </section>
  );
};

export default MyReservations;
