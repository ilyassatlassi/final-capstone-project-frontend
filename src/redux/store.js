import { configureStore } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import userReducer from './slices/user';
import doctorsReducer from './slices/doctors';
import reservationsReducer from './slices/reservations';

export default configureStore({
  reducer: {
    user: userReducer,
    doctors: doctorsReducer,
    reservations: reservationsReducer,
  },
  middleware: [thunk, createLogger()],
});
