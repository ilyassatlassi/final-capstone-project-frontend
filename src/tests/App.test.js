import MyReservations from "../components/MyReservations";
import { Provider } from "react-redux";
import store from '../redux/store';
import renderer from "react-test-renderer"

test("First snapshot test", ()=> {
    const component = renderer.create(
        <Provider store={store}>
             <MyReservations />
        </Provider>
    );
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})



