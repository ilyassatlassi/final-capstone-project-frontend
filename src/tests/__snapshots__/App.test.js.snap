// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First snapshot test 1`] = `
<section
  className="grid lg:grid-cols-3 h-screen gap-4 w-screen justify-center p-6 bg-[#97BFOE]"
/>
`;
